backend/src/test/java/com/now/naaga/letter/application/LetterServiceTest.java

```diff
@@ -0,0 +1,103 @@
+import com.now.naaga.game.exception.GameException;
+import com.now.naaga.letter.application.letterlog.dto.LetterCreateCommand;
+import com.now.naaga.letter.domain.Letter;
+import com.now.naaga.place.domain.Position;
+import com.now.naaga.player.domain.Player;
+import com.now.naaga.player.exception.PlayerException;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+
+import static com.now.naaga.common.fixture.MemberFixture.MEMBER;
+import static com.now.naaga.common.fixture.PlayerFixture.NICKNAME;
+import static com.now.naaga.common.fixture.PlayerFixture.SCORE;
+import static com.now.naaga.common.fixture.PositionFixture.잠실_루터회관_정문_좌표;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@ActiveProfiles("test")
+@Sql("/truncate.sql")
+@SpringBootTest
+class LetterServiceTest {
+    
+    @Autowired
+    private LetterService letterService;
+    
+    @Autowired
+    private PlayerBuilder playerBuilder;
+    
+    @Autowired
+    private GameBuilder gameBuilder;
+    
+    
+    @Test
+    void 쪽지를_정상적으로_생성하고_게임중_등록한_쪽지를_기록으로_남긴다() {
+        //given
+        final Player savedPlayer = playerBuilder.init()
+                .build();
+        final String message = "날씨가 선선해요.";
+        final Position position = 잠실_루터회관_정문_좌표;
+        gameBuilder.init()
+                .player(savedPlayer)
+                .build();
+        
+        //when
+        final LetterCreateCommand letterCreateCommand = new LetterCreateCommand(
+                savedPlayer.getId(),
+                message,
+                position);
+        final Letter expected = letterService.createLetter(letterCreateCommand);
+        
+        //then
+        assertSoftly(softAssertions -> {
+            softAssertions.assertThat(expected.getMessage()).isEqualTo(message);
+            softAssertions.assertThat(expected.getRegisteredPlayer()).isEqualTo(savedPlayer);
+        });
+    }
+    
+    @Test
+    void 쪽지를_등록할_때_현재_진행_중인_게임이_존재하지_않으면_예외가_발생한다() {
+        //given
+        final Player savedPlayer = playerBuilder.init()
+                .build();
+        final String message = "날씨가 선선해요.";
+        final Position position = 잠실_루터회관_정문_좌표;
+        final LetterCreateCommand letterCreateCommand = new LetterCreateCommand(
+                savedPlayer.getId(),
+                message,
+                position);
+        
+        //when&then
+        assertThatThrownBy(() -> letterService.createLetter(letterCreateCommand))
+                .isInstanceOf(GameException.class);
```

Comment:
P3:

코코닥이 저에게 해줬던 리뷰인데요.
```suggestion
        Assertions.assertAll(() -> {
            final BaseExceptionType baseExceptionType = assertThrows(GameException.class, () -> letterService.createLetter(letterCreateCommand))
                    .exceptionType();
            assertThat(baseExceptionType).isEqualTo(GameExceptionType.NOT_EXIST_IN_PROGRESS);
        });
```
이런 식으로 Junit의  assertThorws로 그 안의 타입까지 검증하면 더 정확한 예외 상황 검증이 가능할 것 같습니다.
제가 이것을 assertAll에 다 묶은 이유는 asserThrows로 예외를 받아내는 과정도 예외가 터지지 않으면 저기서 테스트가 터지기 때문에 저 것을 모두 검증의 영역으로 봐서입니다.
assertSoftly를 사용하지 않은 이유는 assertThrows가 assertJ가 아니여서 softAssertions.으로 체이닝이 불가능하더라구요